src/osd/PrimaryLogPG.cc

```diff
@@ -8504,9 +8504,16 @@ void PrimaryLogPG::_do_rollback_to(OpContext *ctx, ObjectContextRef rollback_to,
   map<snapid_t, interval_set<uint64_t> >::iterator iter =
     snapset.clone_overlap.lower_bound(snapid);
   ceph_assert(iter != snapset.clone_overlap.end());
-  interval_set<uint64_t> overlaps = iter->second;
+  auto& last_clone_iter = --snapset.clone_overlap.end();
+
+  interval_set<uint64_t> rollback_overlaps;
```

Comment:
maybe a check of last_clone_iter != snapset.clone_overlap.begin() is needed?

Reply:
last_clone_iter == snapset.clone_overlap.begin() should be fine.  The assert above
```
    map<snapid_t, interval_set<uint64_t> >::iterator iter =
      snapset.clone_overlap.lower_bound(snapid);
    ceph_assert(iter != snapset.clone_overlap.end());
```
proves that snapset.clone_overlap is non-empty non-empty, so snapset.clone_overlap.end() != snapset.clone_overlap.begin() and --snapset.clone_overlap.end() should be valid.  last_clone_iter probably shouldn't be a reference, though, as it outlives the temp and accessing it is UB.